src/parquet/column/properties.h

```diff
@@ -150,77 +164,95 @@ class PARQUET_EXPORT WriterProperties {
     }
 
     std::shared_ptr<WriterProperties> build() {
-      return std::shared_ptr<WriterProperties>(
-          new WriterProperties(allocator_, dictionary_pagesize_, default_encoding_,
-              encodings_, pagesize_, version_, default_codec_, codecs_));
+      return std::shared_ptr<WriterProperties>(new WriterProperties(allocator_,
+          dictionary_enabled_, dictionary_pagesize_, pagesize_, version_, created_by_,
+          default_encoding_, encodings_, default_codec_, codecs_));
     }
 
    private:
     MemoryAllocator* allocator_;
+    bool dictionary_enabled_;
     int64_t dictionary_pagesize_;
+    int64_t pagesize_;
+    ParquetVersion::type version_;
+    std::string created_by_;
     // Encoding used for each column if not a specialized one is defined as
     // part of encodings_
     Encoding::type default_encoding_;
     std::unordered_map<std::string, Encoding::type> encodings_;
-    int64_t pagesize_;
-    ParquetVersion::type version_;
     // Default compression codec. This will be used for all columns that do
     // not have a specific codec set as part of codecs_
     Compression::type default_codec_;
     ColumnCodecs codecs_;
   };
 
-  MemoryAllocator* allocator() const { return allocator_; }
+  inline MemoryAllocator* allocator() const { return allocator_; }
+
+  bool dictionary_enabled() const { return dictionary_enabled_; }
+
+  inline int64_t dictionary_pagesize() const { return dictionary_pagesize_; }
 
-  int64_t dictionary_pagesize() const { return dictionary_pagesize_; }
+  inline int64_t data_pagesize() const { return pagesize_; }
 
-  int64_t data_pagesize() const { return pagesize_; }
+  inline ParquetVersion::type version() const { return parquet_version_; }
 
-  ParquetVersion::type version() const { return parquet_version_; }
+  inline std::string created_by() const { return parquet_created_by_; }
 
-  Encoding::type encoding(const std::shared_ptr<schema::ColumnPath>& path) const {
-    Encoding::type coding = default_encoding_;
+  inline Encoding::type encoding(const std::shared_ptr<schema::ColumnPath>& path) const {
     auto it = encodings_.find(path->ToDotString());
-    if (it != encodings_.end()) { coding = it->second; }
+    if (it != encodings_.end()) { return it->second; }
+    return default_encoding_;
+  }
+
+  inline Encoding::type level_encoding() const { return Encoding::RLE; }
```

Comment:
Do we really want to make this flexible?

Maybe useful when we in future add another level encoding but I do not see the benefit at the moment.